### PR TITLE
Allow Laravel 10

### PR DESCRIPTION
**Pull request by mosbror**
Require PHP ^8.1 as it's required by Laravel 10.
Disallow Larvavel 8.

### CODE REVIEW

**Review by jesperbjerke — CHANGES_REQUESTED** [review]
If there are no changes needed to the code of this package in order to support Laravel 10, there's no need to bump the minimum requirements for either php or the Laravel dependencies. This way, this package will still work with older versions of php and Laravel and there will be no need to publish a major version.
The php req could stay at "^7.3 || ^8.0" because it allows both 7.3+ and everything between 8.0 < 9.0.
The other dependencies could have "^8.0 | ^9.0 | ^10.0".

**Comment by noplanman** [comment]
It does need updating for Laravel 10 though!

To keep backward compatibility, just add Laravel 10 like this:

```diff
-        "illuminate/support": "^8.0 | ^9.0",
-        "illuminate/http": "^8.0 | ^9.0",
-        "illuminate/database": "^8.0 | ^9.0"
+        "illuminate/support": "^8.0 | ^9.0 | ^10.0",
+        "illuminate/http": "^8.0 | ^9.0 | ^10.0",
+        "illuminate/database": "^8.0 | ^9.0 | ^10.0"
```

**Comment by noplanman** [comment]
@jesperbjerke Any chance of getting this in? Or is this library discontinued?

**Comment by jesperbjerke** [comment]
I cannot merge this PR because it would create an unnecessary breaking change. I created a new release: https://github.com/jesperbjerke/laravel-api-query-builder/releases/tag/v1.8.0 which just allows Laravel 10. However, at the time, I am not using this package myself and do not have an environment where I can develop or test this package properly. I am open to review and merge PR's though, or even accept additional maintainers @noplanman

**Pull request closed without merging**
